src/calendar.js

```diff
@@ -42,16 +43,17 @@ export default class Calendar extends Component {
 
   render() {
     const m = this.props.moment;
-    const d = m.date();
-    const d1 = m.clone().subtract(1, 'month').endOf('month').date();
-    const d2 = m.clone().date(1).day();
-    const d3 = m.clone().endOf('month').date();
-    const days = [].concat(
-      range(d1 - d2 + 1, d1 + 1),
-      range(1, d3 + 1),
-      range(1, 42 - d3 - d2 + 1)
-    );
-    const weeks = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
+    const date = m.date();
+
+    const monthStartWeekDay = m.clone().startOf('month').format('e');
+    const monthEndWeekDay = m.clone().endOf('month').format('e');
```

Comment:
These actually return strings, so I'm surprised that the `===` tests below work (although I can attest that they _do_...).

Local-aware weekday as a number can be done via `.weekday()` (as opposed to non-locale `.day()`)

Reply:
`moment().format('e')` returns number
You can change it to `.weekday()` as it's more declarative if you wish

Reply:
`format('e')` -- as per documentation -- returns the weekday number as a String (all format methods return Strings).
When doing Math with that output, it's easy to introduce errors, since `"1" + 1 === "11"`.